integration_test/third_party_apps_data/applications/dcgm/centos_rhel/install

```diff
@@ -1,54 +1,81 @@
+    curl -fSsl -O https://us.download.nvidia.com/tesla/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
+    sudo bash ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --silent
 
-# check NVIDIA driver installation succeeded
-nvidia-smi
+    # check NVIDIA driver installation succeeded
+    nvidia-smi
+}
 
-# Install DCGM
-VERSION_ID=${VERSION_ID%%.*}
-sudo yum-config-manager \
-    --add-repo http://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
-sudo yum clean expire-cache
-sudo yum install -y datacenter-gpu-manager
-sudo systemctl --now enable nvidia-dcgm
-
-# check DCGM service running and load profiling module
-dcgmi discovery --list
+install_dcgm() {
+    # Ref: https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#rhel-centos-rocky-linux
+    sudo yum install -y yum-utils
+    sudo yum-config-manager \
+    --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+    sudo yum clean expire-cache
+    sudo yum install -y datacenter-gpu-manager
+    sudo systemctl --now enable nvidia-dcgm 
+
+    # check DCGM service running and load profiling module
+    dcgmi discovery --list
+}
+
+try_install() {
+    for install_method in "$@"
+    do
+        echo "Installing NVIDIA driver with $install_method..."
+        # Run the installation method in a subshell with -e so that the function
+        # exit on error
+        set +e ; ( set -e; $install_method ) ; exit_code=$?
+        set -e
+        if [ $exit_code -eq 0 ]; then
+            echo "NVIDIA driver has been installed successfully with $install_method."
+            return 
+        fi
+    done
+    return 1
+}
+
+handle_rehl7() {
+    install_driver_from_package_manager() {
+        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=7&target_type=rpm_network
+        sudo yum install -y yum-utils
+        sudo yum-config-manager \
+        --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+        sudo yum clean all
+        sudo yum -y install nvidia-driver-latest-dkms
+
+        # check NVIDIA driver installation succeeded
+        nvidia-smi
+    }
+    try_install install_driver_from_runfile install_driver_from_package_manager
+}
+
+handle_common() {
+    install_driver_from_package_manager() {
+        # Ref: https://developer.nvidia.com/cuda-12-2-2-download-archive?target_os=Linux&target_arch=x86_64&Distribution=RHEL&target_version=8&target_type=rpm_network
+        sudo dnf config-manager \
+        --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel$VERSION_ID/x86_64/cuda-rhel$VERSION_ID.repo
+        sudo dnf clean all
+        sudo dnf -y module install nvidia-driver:latest
+
+        # check NVIDIA driver installation succeeded
+        nvidia-smi
+    }
```

Comment:
This is _almost_ identical to `install_driver_from_package_manager` defined in `handle_rehl7`[sic]. The only differences are `s/yum/dnf` and `module install nvidia-driver:latest` vs `install nvidia-driver-latest-dkms`. Is there anything we can do to share more of the code (e.g., given that `yum` and `dnf` can pretty much be used interchangeably on newer systems)?